api/ordinals.ts

```diff
@@ -47,48 +42,45 @@ const sortOrdinalsByConfirmationTime = (prev: BtcOrdinal, next: BtcOrdinal) => {
   return 0;
 };
 
-export async function fetchBtcOrdinalsData(
-  btcAddress: string,
-  network: NetworkType
-): Promise<BtcOrdinal[]> {
+export async function fetchBtcOrdinalsData(btcAddress: string, network: NetworkType): Promise<BtcOrdinal[]> {
   const btcClient = new BitcoinEsploraApiProvider({
     network,
   });
-  const unspentUTXOS = await btcClient.getUnspentUtxos(btcAddress);
+  const addressUTXOs = await btcClient.getUnspentUtxos(btcAddress);
   const ordinals: BtcOrdinal[] = [];
+
   await Promise.all(
-    unspentUTXOS.map(async (utxo: UTXO) => {
-      const ordinalContentUrl = `${XVERSE_API_BASE_URL}/v1/ordinals/output/${utxo.txid}/${utxo.vout}`;
-      try {
-        const ordinal = await axios.get(ordinalContentUrl);
-        if (ordinal) {
-          ordinals.push({
-            id: ordinal.data.id,
-            confirmationTime: utxo.status.block_time || 0,
-            utxo,
+    addressUTXOs
+      .filter((utxo) => utxo.status.confirmed) // we can only detect ordinals from confirmed utxos
+      .map(async (utxo: UTXO) => {
+        const ordinalContentUrl = `${XVERSE_INSCRIBE_URL}/v1/inscriptions/utxo/${utxo.txid}/${utxo.vout}`;
+
+        const ordinalIds = await axios.get<string[]>(ordinalContentUrl);
```

Comment:
we should replace this block of code with the `getOrdinalIdFromUtxo` function

Reply:
They do slightly different things. getOrdinalIdFromUtxo returns only one ID on a UTXO because that's what the original method did. This one gets IDs of all ordinals on a UTXO if there are multiple. We can rewrite getOrdinalIdFromUtxo to return multiple IDs as another issue where we'd have to update the extension and mobile app code appropriately.